### ISSUE TITLE
Mock always calls __destruct (PHP 5.4/5.5, any Mockery version)

### ISSUE THREAD

**Issue by tiraeth** [mockery/mockery#355]
Hi,

I've encountered an error with `__destructor` being called regardless constructor call setup. See code below:

``` php
<?php

require_once __DIR__ . '/vendor/autoload.php';

class Foo
{
    public function __construct()
    {
        echo 'Foo::__construct' . PHP_EOL;
    }

    public function __destruct()
    {
        echo 'Foo::__destruct' . PHP_EOL;
    }

    public function bar()
    {
    }
}

// ...
```

First attempt (incorrect behaviour):

``` php
<?php

// ...

$fooMock = Mockery::mock('Foo');

Mockery::close();
```

Expected output: **(empty)**
Received output: **Foo::__destruct\n**

Second attempt (correct behaviour):

``` php
<?php

// ...

$fooMock = Mockery::mock('Foo[bar]');

Mockery::close();
```

Expected output: **Foo::__construct\nFoo::__destruct\n**
Received output: **Foo::__construct\nFoo::__destruct\n**

Environment: PHP `5.5.14`, Mockery `dev-master` (and `0.9.1`), doesn't matter if I have XDebug enabled or not.

I've found that `__destruct` is blacklisted in `$builder` but still I'm curious why is it called. Mockery or PHP bug?

Appreciate your help, thanks!

**Edit:** Checked on OS X's bundled PHP `5.4.24` and got the same output.

**Edit 2:** Checked `0.9.0`, `0.8.0`, and `0.7.2` versions of mockery. Still the same result.

**Comment by antonienko** [comment]
Any news on this issue?

**Comment by robertbasic** [comment]
This was implemented by #215.
